pkg/util/pki/common.go

```diff
@@ -94,10 +94,13 @@ type UserCertificate struct {
 }
 
 // DN returns the Distinguished Name of a TLS certificate
-func (u *UserCertificate) DN() string {
+func (u *UserCertificate) DN() (string, error) {
 	// cert has already been validated so we can assume no error
-	cert, _ := certutil.DecodeCertificate(u.Certificate)
-	return cert.Subject.String()
+	cert, err := certutil.DecodeCertificate(u.Certificate)
+	if err != nil {
+		return "", err
```

Comment:
**Question**: wouldn't it be better to wrap the error with some content information (we are doing this in KafkaUser DistinguishedName retrieval), IMO it would help with debug clarification.
(These cases I always have the `os.Open()` errors in mind where the non-existing path object error is `no such file or directory`, but the path itself is not included - although that would be a detail, not a wrapping, but I mean the contextual information in general).

Reply:
I'm not sure if any useful information is available here that could be added to the error context. The caller can add more details to the error like the `KafkaUser` custom resources name (https://github.com/banzaicloud/koperator/pull/843/files#diff-2b2cc6c6efb0a3a4b49078c08a64ff936f6d02a73543d0c8f98a7b9b0c4f2a2eR243)

Reply:
Yeah, I meant an error message to narrow the context from generic certificate error to KafkaUser DistringuishedName determination certificate error

Reply:
The error message already contains the kafkauser name, namespace.

Reply:
Reconciler dumps the stacktrace.